src/roman_datamodels/datamodels/_datamodels.py

```diff
@@ -77,55 +77,37 @@ def append_individual_image_meta(self, meta):
 
             # Keys that are themselves Dnodes (subdirectories)
             # neccessitate a new table
+            if isinstance(value, (dict, asdf.tags.core.ndarray.NDArrayType, QTable)):
+                continue
+
             if isinstance(value, stnode.DNode):
                 # Storage for keys and values
                 subtable_cols = []
                 subtable_vals = []
 
                 # Loop over items within the node
                 for subkey, subvalue in meta_dict[key].items():
-                    # Skip ndarrays
-                    if isinstance(subvalue, asdf.tags.core.ndarray.NDArrayType):
-                        continue
-                    # Skip QTables
-                    if isinstance(subvalue, QTable):
+                    # Skip ndarrays and QTables
+                    if isinstance(subvalue, (asdf.tags.core.ndarray.NDArrayType, QTable)):
                         continue
 
                     subtable_cols.append(subkey)
-
-                    if isinstance(subvalue, (list, dict)):
-                        subtable_vals.append([str(subvalue)])
-                    else:
-                        subtable_vals.append([subvalue])
+                    subtable_vals.append([str(subvalue)] if isinstance(subvalue, (list, dict)) else [subvalue])
 
                 # Skip this Table if it would be empty
-                if len(subtable_vals) == 0:
-                    continue
-
-                # Make new Keyword Table if needed
-                if (key not in self.meta.individual_image_meta) or (self.meta.individual_image_meta[key].colnames == ["dummy"]):
-                    self.meta.individual_image_meta[key] = QTable(names=subtable_cols, data=subtable_vals)
-                else:
-                    # Append to existing table
-                    self.meta.individual_image_meta[key].add_row(subtable_vals)
-            # Skip non-schema metas
```

Comment:
Shouldn't we keep these checks for non DNodes?

Reply:
They were hoisted to the beginning of the block:
```
if isinstance(value, (dict, asdf.tags.core.ndarray.NDArrayType, QTable)):
                continue
```